modules/core/README.md

```diff
@@ -235,22 +247,85 @@ Replace `xxx` with the instance ID.
 
 ### Upgrading Nomad Clients
 
-When draining Nomad client nodes, users will experience momentary downtime as ELB catches up with
-the unhealthy client status.
+**Important**: These steps are recommended to minimise the outage your services might experience. In particular,
+if you service only has one instance of it running, you will definitely encounter outage. Ensure
+that your services have at least two instances running.
 
-1. Drain the servers using `nomad node-drain node-id`
-1. After the allocations are drained, terminate the instance and AWS will launch a new instance.
+1. Build your new AMI, and Terraform apply the new AMI.
+2. Take note of the old instances ID that you are going to retiring. You can get a list of the instance IDs with the command:
 
-You can use this AWS CLI command:
+```bash
+aws autoscaling describe-auto-scaling-groups \
+    --auto-scaling-group-name ASGName \
+    | jq --raw-output '.AutoScalingGroups[0].Instances[].InstanceId' \
+    | tee instance-ids.txt
+```
+
+3. Using Terraform or the AWS console, set the `desired` capacity of your auto-scaling group to twice the current desired value. Make sure the `maximum` is set to a high enough value so that you set the appropriate `desired` value. This is spin up new clients that will be prepared to take over from the instances you are retiring. Alternatively, you can use the [AWS CLI](https://docs.aws.amazon.com/cli/latest/reference/autoscaling/update-auto-scaling-group.html) too:
```

Comment:
Nitpick: `This is spin up new clients that will be prepared to`, this might be better: `This spins up new clients that are prepared to`?